### ISSUE TITLE
LSP is down after upgrading to Spyder-5.1.0 even when downgrading back to 5.0.5

### ISSUE THREAD

**Issue by yaront** [spyder-ide/spyder#16116]
I upgraded to 5.1.0 on MacOs (direct download). It said that there was an error in opening Spyder due to the Spyder-kernels (asked me to download 2.1.0). I couldn't download that version (only 2.0.5), so I restored spyder-5.0.5 (either from the Time Machine and as clean download from the website), but since then the LPS is failing and I get the following message:

![image](https://user-images.githubusercontent.com/39324784/127818811-1e1baf9d-a9cf-4392-8db0-dce6816d4c1b.png)

* Spyder version: 5.0.5
* Python version: 3.7.10 (conda env)
* Qt version: 5.12.10
* PyQt version: 5.12.3
* Operating System name/version: MacOS BigSur 11.5.1

### Dependencies

```
# Mandatory:
applaunchservices >=0.1.7     :  0.2.1 (OK)
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.24.1 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=1.0                  :  2.8.3 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.7 (OK)
pyls_spyder >=0.3.2;<0.4.0    :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.0 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.1.0 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=39.0.0           :  57.0.0 (OK)
sphinx >=0.6.6                :  4.0.2 (OK)
spyder_kernels >=2.0.4;<2.1.0 :  2.0.4 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.2 (OK)
zmq >=17                      :  22.1.0 (OK)

# Optional:
cython >=0.21                 :  0.29.23 (OK)
matplotlib >=2.0.0            :  3.4.2 (OK)
numpy >=1.7                   :  1.21.0 (OK)
pandas >=1.1.1                :  1.2.5 (OK)
scipy >=0.17.0                :  1.7.0 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

**Comment by AndrewAmmerlaan** [comment]
Probably this is due to that spyder-5.1.0 has switched from `python-language-server`, to `python-lsp-server`. After the upgrade your spyder configuration is likely trying to start `pylsp` instead of `pyls`. You can try installing `python-lsp-server` if it is not already installed. Or use the reverse workaround from Comment 13 on this Bug: https://bugs.gentoo.org/783615#c13

**Comment by ccordoba12** [comment]
Hey @yaront, thanks for reporting. I have two questions for you:

* How did you update to Spyder 5.1.0, since it's not available in Anaconda nor conda-forge?
* Did you update Spyder in a separate conda environment or in you `base` one?

**Comment by yaront** [comment]
Hi @ccordoba12 !

> * How did you update to Spyder 5.1.0, since it's not available in Anaconda nor conda-forge?

Downloaded it directly from the website.

> * Did you update Spyder in a separate conda environment or in you `base` one?

Again, it is a standalone installation, not through conda. But the interpreter is pointed to the separated environment (with Python 3.7), not `base`.

When I ran the 5.1.0 I got an error message, saying I should upgrade `Spyder-kernels` to 2.1.0. Not sure why, but I was not managed to do that (neither through conda nor through pip)...

According to @AndrewAmmerlaan comment, after downgrading back to 5.0.5, the "Module for the Python language server" was still set to `pylsp` instead of `pyls` (probably due to the former installation of 5.1.0). Setting it back to `pyls` solved the problem.

**Comment by ccordoba12** [comment]
Ok, I see. Thanks for your answers. Probably this error is related to issue #16117, so it could be fixed in our 5.1.1 version, which I hope to release in some hours.

**Comment by ccordoba12** [comment]
I'll reopen this so we don't forget to do a followup with you @yaront.

**Comment by mrclary** [comment]
@yaront, the LSP should start working again if you first delete your `./.spyder-py3` directory (or rename it), then startup Spyder 5.0.5. If you have a number of preferences set and you renamed the directory instead of removing it, you should be able to restore them manually from the preference files in the old configuration directory.

**Comment by mrclary** [comment]
@yaront, alternatively, you can open `~/.spyder-py3/config/spyder.ini` and find the line that starts with `provider_configuration =` and replace all occurrences of `'pylsp'` with `'pyls'`.

**Comment by dkmj** [comment]
following @mrclary tip, deleted .spyder-py3 and am up again on 5.0.5. 5.1 crashed for me as for a few others, siting lacking kernals package (same error as with initial 5.0 release?) awaiting standalone 5.1.1 :-)

**Comment by ccordoba12** [comment]
Hey everyone, I just released 5.1.1 an hour ago. Please try it and let us know if it works for you.

**Comment by ccordoba12** [comment]
I'm going to close this one since @mrclary confirmed all issues with our Mac installer were solved in 5.1.1
